### ISSUE TITLE
Document variant decision

### ISSUE THREAD

**Issue by mkoennecke** [nexusformat/definitions#645]
Hi,

following up on a recent question posed to me I found out that at NIAC 2014 we decided on a scheme 
for encoding variants of a value in NeXus. The scheme looks like this(I believe):

distance
    @variant=distance_from_reading
distance_from_reading
    @variant=distance_as_guessed_by_cleaner
distance_as_guessed_by_cleaner

I.e. a value for which a better value becomes available gets a variant attribute pointing to improved 
value. The name of which is built from the original name plus a description appended with underscores. The whole can become a chain of some length.

I had to dig this information up from Telco minutes. The decision ism actually in the NIAC2014_Minutes.md but without details. But this is not in the manual.

Thus I suggest:
- Please someone check that this is correct: (Herbert: it came from the concordance discussions with CIF)
- Then put this in the manual, probably in the section describing storage of individual data items in the file.

**Comment by yayahjb** [comment]
Variants in CBF are a little different, since we are dealing with the
tables of a relational database instead of a tree, so in that case, variant
is
a column in any table and the name of the table does not have to change, a
variant of a row is just another row.  The scheme Mark proposes is a
natural way to accomplish a similar result without having relations.

Regards,
    Herbert

**Comment by rayosborn** [comment]
I don't recall that discussion, although I assume I was there. I am concerned about the idea that the variant is the improved value, because it would mean that the value stored in the NXDL-approved field is not the best one to use, even though it is the value that will be used by any analysis code. The value in a standard field like 'distance' should, in my view, always be the value that analysis software is expected to use. It also seems an impractical naming scheme. Trying to add what is effectively additional metadata to the name seems like a bad idea. I hope I didn't vote for it, but I agree that we shouldn't just ignore previous votes.

**Comment by rayosborn** [comment]
It's the last point I find worrying. Someone writing a third-party software package, not knowing anything about a particular experiment, is going to use the NXDL-approved field. If we discover, after the experiment, that we got the distance wrong, we have to change that value, not create another undocumented field to store it. I'm not against storing variants even if I don't like the naming scheme; I'm just against stating that they contain "improved" values.

**Comment by vasole** [comment]
I am with @rayosborn on this one. If an analysis code following a definition has to look for variants of the definition instead of the expected fields this thing is simply wrong. There are alternative solutions like writing new files and/or definitions just with links so that space is not a concern. Furthermore, if one wants to indicate something has changed, it is preferable to overwrite the "bad" value and create whatever variant one might need to specify what has been changed.

**Comment by mkoennecke** [comment]
Hi,

here again the exit information I have. On the last Telco, as of February, 7,2019 we decided to go with this.

This is what I found in my Telco notes for the week 14 2014. 

-------------------------------------------------------------------------------------------------------------------------------------------------------
Variants
---------

It was aggreed to treat variants in a similar way as we treat
errors. Thus the actual field with the NeXus name will always hold the
best value. And in an attribute the names of any variants for the
field. Example:

distance
   @variant=distance_from_reading
distance_from_reading
   @variant=distance_as_guessed_by_cleaner
distance_as_guessed_by_cleaner

To be ratified at NIAC.

----------------------------------------------------------------------------------------------------------------------------------------------
And in the NIAC 2014 notes I found that this was actually ratified.
